### DISCUSSION TITLE
Google analytics

### DISCUSSION THREAD

**Discussion by NanoscaleSimulations** [reflex-dev/reflex#1800] [Q&A]
Anyone succesfully set up google analytics for their reflex app/website?
Or does anyone know how to do it?

Or is there another way to track visitors and interactions?

Thanx in advance :)

**Comment by picklelo** [comment]
Hey @NanoscaleSimulations, you should be able to do this by inserting the `rx.script` necessary into your app.

```python
def index():
    return container(
        rx.script(
            src="https://www.googletagmanager.com/gtag/js?id=<TAG>",
            strategy="afterInteractive",
        ),
        rx.script(
            """
            window.dataLayer = window.dataLayer || [];
            function gtag(){window.dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', <TAG>);
            """,
            id="google-analytics",
            strategy="afterInteractive",
        ),
        # Rest of the page contents here
    )
```

You would have to add this to every page, but you can abstract it out into a template that you apply to all your pages.
